ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-popover.js

```diff
@@ -0,0 +1,243 @@
+      editingInGwei,
+      setMaxBaseFeeGWEI,
+      setMaxBaseFeeMultiplier,
+    ],
+  );
+
+  const updateBaseFeeMode = useCallback(
+    (value) => {
+      if (value) {
+        const baseFeeGwei = multiplyCurrencies(
+          estimatedBaseFee,
+          maxBaseFeeMultiplier,
+          {
+            numberOfDecimals: 6,
+            multiplicandBase: 16,
+            multiplierBase: 10,
+            toNumericBase: 'dec',
+            fromDenomination: 'WEI',
+            toDenomination: 'GWEI',
+          },
+        );
+        setMaxBaseFeeGWEI(baseFeeGwei);
+      } else {
+        const baseFeeMul = divideCurrencies(
+          decGWEIToHexWEI(maxBaseFeeGWEI),
+          estimatedBaseFee,
+          {
+            numberOfDecimals: 6,
+            dividendBase: 16,
+            divisorBase: 16,
+            toNumericBase: 'dec',
+          },
+        );
+        setMaxBaseFeeMultiplier(baseFeeMul);
+      }
+      setEditingInGwei(value);
+    },
+    [
+      estimatedBaseFee,
+      maxBaseFeeMultiplier,
+      maxBaseFeeGWEI,
+      setMaxBaseFeeGWEI,
+      setMaxBaseFeeMultiplier,
+      setEditingInGwei,
+    ],
+  );
+
+  return (
+    <Popover
+      className="advanced-gas-fee-popover"
+      title={t('advancedGasFeeModalTitle')}
+      onBack={() => onClose()}
+      onClose={() => onClose()}
+    >
+      <Box className="advanced-gas-fee-popover" margin={4}>
+        <FormField
+          onChange={(value) => {
+            onManualChange?.();
+            setBaseFee(value);
+          }}
+          titleFontSize={TYPOGRAPHY.H7}
+          titleText={t('maxBaseFee')}
+          titleUnit={
+            editingInGwei ? t('gweiInParanthesis') : t('multiplierInParathesis')
+          }
+          tooltipText={t('advancedBaseGasFeeToolTip')}
+          titleDetail={
+            editingInGwei ? (
+              <Button
+                className="advanced-gas-fee-popover__edit-link"
+                type="link"
+                onClick={() => updateBaseFeeMode(false)}
+              >
+                <I18nValue messageKey="editInMultiplier" />
+              </Button>
+            ) : (
+              <Button
+                className="advanced-gas-fee-popover__edit-link"
+                type="link"
+                onClick={() => updateBaseFeeMode(true)}
+              >
+                <I18nValue messageKey="editInGwei" />
+              </Button>
+            )
+          }
+          value={editingInGwei ? maxBaseFeeGWEI : maxBaseFeeMultiplier}
+          detailText={
+            editingInGwei
+              ? t('baseFeeFiatFromMultiplier', [
+                  maxBaseFeeMultiplier,
+                  baseFeeFiatMessage,
+                ])
+              : t('baseFeeFiatFromGwei', [baseFee, baseFeeFiatMessage])
+          }
+          numeric
+          bottomBorder
+          inputDetails={
+            <AdvancedGasFeeInputSubtext
+              currentData={`${estimatedBaseFeeInDecGWEI} GWEI`}
```

Comment:
Instead of calling this `current` vs `twelveHr`, what do we think about naming this `latest` vs. `historical`? This way if we change the time range we don't have to update the naming. I am also using this same naming on the `controller` side (and so is Sam on the API side).

Reply:
Good point, I did the change